JavaScript/JavaScriptSDK/Telemetry/PageViewPerformance.ts

```diff
@@ -137,12 +146,31 @@ module Microsoft.ApplicationInsights.Telemetry {
         }
 
         public static getDuration(start: any, end: any): number {
-            var duration = 0;
+            var duration = undefined;
             if (!(isNaN(start) || isNaN(end))) {
                 duration = Math.max(end - start, 0);
             }
 
             return duration;
         }
+
+        /**
+         * GoogleBot is returning invalid values in performance.timing API.
+         * This method tells if given durations should be excluded from collection.
+         */
+        public static shouldCollectDuration(...durations: number[]): boolean {
+            let userAgent = navigator.userAgent;
+            let isGoogleBot = userAgent ? userAgent.toLowerCase().indexOf("googlebot") !== -1 : false;
+
+            if (isGoogleBot) {
```

Comment:
I still want to report page duration from GoogleBot if it gives accurate numbers. Let's keep the rule more strict for now. We can revisit it after a week if it's not enough.